billboard.py

```diff
@@ -273,3 +282,18 @@ def getPositionRowValue(rowName):
 
             entry = ChartEntry(title, artist, peakPos, lastPos, weeks, rank, isNew)
             self.entries.append(entry)
+
+def list_all_charts():
+    req = requests.get('https://www.billboard.com/charts', headers=HEADERS, timeout=25)
+    soup = BeautifulSoup(req.text, 'html.parser')
+    chartCategories = soup.find('div', {'class' : "chart-panel--main"}).findAll('div', {'class' : "chart-panel__text"})
+    chartGroups = soup.findAll('div', {'class' : "chart-panel__charts"})
+    sFull = ""
+    for category, group in zip(chartCategories, chartGroups):
+        s = category.text.strip()
+        s += '\n' + '-' * len(s) + '\n'
+        charts = group.findAll('a', {'class' : "chart-panel__link"})
+        for c in charts:
+            s += c['href'].split('/')[-1] + '\n'
+        sFull += s + '\n'
+    return sFull
```

Comment:
The current output is very human-readable:

```
Top Charts
----------
hot-100
billboard-200
artist-100
...

Greatest of All Time
--------------------
greatest-billboard-200-albums
greatest-hot-100-singles
greatest-hot-100-artists
...
```

But I would rather the output be easily machine-readable, e.g., a list of strings:

```
['hot-100', 'billboard-200', ...]
```

But that loses the category information, so maybe a dict mapping to lists of strings?

```
{'Top Charts': ['hot-100', 'billboard-200', ...], 'Greatest of All Time': ['greatest-billboard-200-albums', ...]}
```

But that seems too complicated. With a flat list of strings, you can easily get a random chart, search for a chart containing `country`, etc.

@jameswenzel: Any thoughts?